src/test/run-pass/nonzero.rs

```diff
@@ -0,0 +1,16 @@
+// Copyright 2018 The Rust Project Developers. See the COPYRIGHT
+// file at the top-level directory of this distribution and at
+// http://rust-lang.org/COPYRIGHT.
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+fn main() {
+    use std::num::NonZeroU32;
+    let nz = NonZeroU32::new(5).unwrap();
+    let num: u32 = nz.into();
```

Comment:
nit: perhaps just make this a doctest on the impl, so users can see it?

Reply:
most of the types like `String`/`PathBuf` in std doesn't have specific docs for their `From` implementation, and invoking `.from/.into` is kind of intuition, so I think that's fine without doctest? but if it's really needed, I'd add it.

Reply:
Definitely not a _need_.  I just found a run-pass test odd for this, as `x.py test src/libcore` doesn't run those.  `src/libcore/tests/nonzero.rs` might also be a reasonable place for the test.

Reply:
`src/libcore/tests/nonzero.rs` also makes sense to me.